src/tests/ftest/util/mdtest_utils.py

```diff
@@ -0,0 +1,216 @@
+        self.stonewall_statusfile = MdtestParam("-x {}") # num of iter of the
+                                                       # creation phase,
+                                                       # StoneWallingStatusFile
+        self.depth = MdtestParam("-z {}")              # depth of hierarchical
+                                                       # directory structure
+
+        self.daos_pool_uuid = MdtestParam("--daos.pool {}") # pool uuid
+        self.daos_svcl = MdtestParam("--daos.svcl {}")      # pool svcl
+        self.daos_cont = MdtestParam("--daos.cont {}")      # cont uuid
+        self.daos_group = MdtestParam("--daos.group {}")    # server group
+        self.daos_chunk_size = MdtestParam(" --daos.chunk_size {}") # chunk
+                                                                    # size
+        self.daos_oclass = MdtestParam("--daos.oclass {}")  # object class
+
+        self.dfs_pool_uuid = MdtestParam("--dfs.pool {}") # pool uuid
+        self.dfs_svcl = MdtestParam("--dfs.svcl {}")      # pool svcl
+        self.dfs_cont = MdtestParam("--dfs.cont {}")      # cont uuid
+        self.dfs_group = MdtestParam("--dfs.group {}")    # server group
+
+    def __str__(self):
+        """Return a MdtestCommand object as a string.
+        Returns:
+            str: the mdtest command with all the defined parameters
+        """
+        params = []
+        for value in self.__dict__.values():
+            value_str = str(value)
+            if value_str != "":
+                params.append(value_str)
+        return " ".join(["mdtest"] + sorted(params))
+
+    def set_params(self, test, path="/run/mdtest/*"):
+        """Set values for all of the mdtest command params using a yaml file.
+        Args:
+            test (Test): avocado Test object
+            path (str, optional): yaml namespace. Defaults to "/run/mdtest/*"
+        """
+        for name, mdtest_param in self.__dict__.items():
+            mdtest_param.set_yaml_value(name, test, path)
+
+    def get_launch_command(self, basepath, processes, hostfile, runpath=None):
+        """Get the process launch command used to run mdtest
+        Args:
+            basepath (str): DAOS base path
+            processes (int): number of host processes
+            hostfile (str): file defining host names and slots
+        Returns:
+            str: returns mdtest command
+        """
+        with open(os.path.join(basepath, ".build_vars.json")) as afile:
+            build_paths = json.load(afile)
+        attach_info_path = os.path.join(basepath, "install/tmp")
+
+        env = {
+            "CRT_ATTACH_INFO_PATH": attach_info_path,
+            "MPI_LIB": "",
+            "DAOS_SINGLETON_CLI": 1,
+        }
+        export_cmd = [
+            "export {}={}".format(key, val) for key, val in env.items()]
+        mpirun_cmd = [
+            "".join([runpath if runpath else build_paths["OMPI_PREFIX"],
```

Comment:
I'd also pass in the mpirun path+command into this method and let the caller decide which path to use.